backend/python/transformers/transformers_server.py

```diff
@@ -186,15 +224,43 @@ def Predict(self, request, context):
         if request.Tokens > 0:
             max_tokens = request.Tokens
 
-        inputs = self.tokenizer(request.Prompt, return_tensors="pt").input_ids
+        inputs = self.tokenizer(request.Prompt, return_tensors="pt")
         if self.CUDA:
             inputs = inputs.to("cuda")
-        if XPU:
+        if XPU and self.OV == False:
             inputs = inputs.to("xpu")
-
-        outputs = self.model.generate(inputs,max_new_tokens=max_tokens, temperature=request.Temperature, top_p=request.TopP, do_sample=True, pad_token_id=self.tokenizer.eos_token_id)
-        generated_text = self.tokenizer.batch_decode(outputs[:, inputs.shape[1]:], skip_special_tokens=True)[0]
-
+            streaming = False
+
+        if streaming:
+            streamer=TextIteratorStreamer(self.tokenizer,
+                                        skip_prompt=True,
+                                        skip_special_tokens=True)
+            config=dict(inputs,
+                        max_new_tokens=max_tokens, 
+                        temperature=request.Temperature, 
+                        top_p=request.TopP,
+                        top_k=request.TopK, 
+                        do_sample=True,
+                        attention_mask=inputs["attention_mask"],
+                        eos_token_id=self.tokenizer.eos_token_id,
+                        pad_token_id=self.tokenizer.eos_token_id,
+                        streamer=streamer)
+            thread=Thread(target=self.model.generate, kwargs=config)
+            thread.start()
+            generated_text = ""
+            for new_text in streamer:
+                generated_text += new_text
+                yield backend_pb2.Reply(message=bytes(new_text, encoding='utf-8'))
+        else:
+            outputs = self.model.generate(inputs["input_ids"],
+                                max_new_tokens=max_tokens, 
+                                temperature=request.Temperature, 
+                                top_p=request.TopP,
+                                top_k=request.TopK, 
+                                do_sample=True,
+                                pad_token=self.tokenizer.eos_token_id)
+            generated_text = self.tokenizer.batch_decode(outputs[:, inputs["input_ids"].shape[1]:], skip_special_tokens=True)[0]
+            yield backend_pb2.Reply(message=bytes(new_text, encoding='utf-8'))
```

Comment:
Should this line only be in the streaming branch above?

Reply:
Hi, 
thank you for reviewing. Sorry I'm not sure about which line: you mean 263 with the `yeld`?
I had to insert it because `intel_extension_for_transformers` seems not to support `TextIteratorStreamer` for streaming generated tokens. 

This is why for XPU inference streaming is disabled and goes to the `false` branch.
The `yeld` here is to support this use case when `Predict()` is called by `PredictStream()`.

Reply:
thanks for contributing this, hopefully I'll have some hardware to run it on next week.

Anyway all I am picking up on is the `new_text` variable which is not defined AFAICT outside the `for` loop. So I wonder if I'm missing the definition or it's undefined, but Python handles it and/or it is redundant?

Reply:
good catch, looks like a leftover

```suggestion
```

It does not need to yield or return, as it would return otherwise `generated_text`